### PR TITLE
add GET and POST queries for user and pin

### PR DESCRIPTION
**Pull request by sokolikp**

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -7,6 +7,14 @@ target 'Sweeper' do
 
   # Pods for Sweeper
   pod 'Parse'
+  pod 'ParseLiveQuery'
   pod 'GoogleMaps'
   pod 'GooglePlaces'
+  post_install do |installer|
+      installer.pods_project.targets.each do |target|
+          target.build_configurations.each do |config|
+              config.build_settings['SWIFT_VERSION'] = '3.2'
```
Comment (wumingx89) [inline Podfile:16]: I'm just curious why we need this. Is it for the live query?

File: Sweeper/AppDelegate.swift [hunk 2]
```diff
@@ -18,13 +18,25 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        
         let configuration = ParseClientConfiguration {
             $0.applicationId = "com.team11.Sweeper"
             $0.server = "http://165.227.6.232:1337/parse"
+            $0.isLocalDatastoreEnabled = true
         }
+        
         Parse.initialize(with: configuration)
         
+        User.getStoredUser(completion: { (user: User?) in
```
Comment (wumingx89) [inline Sweeper/AppDelegate.swift:30]: can we check User.currentuser like we did for Twitter? I think that's a better pattern to use.

File: Sweeper/Storyboards/Pinviews.storyboard [hunk 3]
```diff
@@ -31,7 +31,7 @@
                 <navigationController storyboardIdentifier="PinviewsNavigationController" automaticallyAdjustsScrollViewInsets="NO" id="MW1-yk-qsy" sceneMemberID="viewController">
                     <toolbarItems/>
                     <navigationBar key="navigationBar" contentMode="scaleToFill" insetsLayoutMarginsFromSafeArea="NO" id="iiy-8k-ooc">
-                        <rect key="frame" x="0.0" y="20" width="375" height="44"/>
+                        <rect key="frame" x="0.0" y="20" width="320" height="44"/>
```
Comment (wumingx89) [inline Sweeper/Storyboards/Pinviews.storyboard:34]: I don't think there any changes to this file worth committing. Not sure why the size changed for you.

**Review by wumingx89 — COMMENTED** [review]
Looks good.

File: Podfile (hunk 1 above)
Comment (sokolikp) [reply Podfile:16]: Yea, LiveQuery breaks without it.

File: Sweeper/Storyboards/Pinviews.storyboard (hunk 3 above)
Comment (sokolikp) [reply Sweeper/Storyboards/Pinviews.storyboard:34]: Hmm yea not sure. I can discard these.

File: Sweeper/AppDelegate.swift (hunk 2 above)
Comment (sokolikp) [reply Sweeper/AppDelegate.swift:30]: Yea but this is a blocking call for Parse, so it's better to do it in a closure. We'll only call this once in appdelegate and from then on we can access via User.currentUser (once it's fetched).

Comment (sokolikp) [reply Sweeper/AppDelegate.swift:30]: Swift gives errors about blocking the main thread if we don't do it in a callback.

**Review by huw30 — APPROVED** [review]

File: Sweeper/AppDelegate.swift (hunk 2 above)
Comment (wumingx89) [reply Sweeper/AppDelegate.swift:30]: Gotcha, sounds good!

**Pull request merged**